projects/dotcms-field-elements/src/components/dot-key-value/dot-key-value.e2e.ts

```diff
@@ -0,0 +1,158 @@
+import { EventSpy } from '@stencil/core/dist/declarations';
+
+describe('dot-key-value', () => {
+    let page: E2EPage;
+    let element: E2EElement;
+    let spyStatusChangeEvent: EventSpy;
+    let spyValueChangeEvent: EventSpy;
+
+    describe('With data', () => {
+        beforeEach(async () => {
+            page = await newE2EPage();
+
+            await page.setContent(`
+            <dot-key-value
+                name="testName"
+                label="testLabel"
+                field-type="Key-Value"
+                hint="testHint"
+                key-placeholder="Enter Key"
+                value-placeholder="Enter Value"
+                value="valueA|1"
+                required-message="testErrorMsg"
+                required="true"
+                save-btn-label="Save"
+                >
+            </dot-key-value>`);
+            element = await page.find('dot-key-value');
+        });
+
+        it('renders', async () => {
+            // tslint:disable-next-line:max-line-length
+            const expectedMarkup = `<dot-key-value name=\"testName\" label=\"testLabel\" field-type=\"Key-Value\" hint=\"testHint\" key-placeholder=\"Enter Key\" value-placeholder=\"Enter Value\" value=\"valueA|1\" required-message=\"testErrorMsg\" required=\"true\" save-btn-label=\"Save\" class=\"dot-valid dot-pristine dot-untouched dot-required hydrated\"><div class=\"dot-field__label\"><label for=\"testName\">testLabel</label><span class=\"dot-field__required-mark\">*</span></div><input id=\"testName\" name=\"key\" placeholder=\"Enter Key\" type=\"text\"><input name=\"value\" placeholder=\"Enter Value\" type=\"text\"><button class=\"dot-key-value__save__button\" type=\"button\">Save</button><key-value-table class=\"hydrated\"><table><tbody><tr><td><button type=\"button\" id=\"valueA_1_0\" class=\"dot-key-value__delete__button\"><label for=\"valueA_1_0\">Delete</label></button></td><td>valueA</td><td>1</td></tr></tbody></table></key-value-table><span class=\"dot-field__hint\">testHint</span></dot-key-value>`;
+            const hint = await element.find('.dot-field__hint');
+            const table = await element.find('table');
+            expect(element.outerHTML).toBe(expectedMarkup);
+            expect(hint).toBeTruthy();
+            expect(table).toBeTruthy();
+            expect(element.classList.contains('dot-valid')).toBe(true);
+            expect(element.classList.contains('dot-pristine')).toBe(true);
+            expect(element.classList.contains('dot-untouched')).toBe(true);
+            expect(element.classList.contains('dot-required')).toBe(true);
+        });
+
+        it('should not render hint', async () => {
+            element.setProperty('hint', '');
+            await page.waitForChanges();
+            const hint = await element.find('.dot-field__hint');
+            expect(hint).toBeNull();
+        });
+
+        it('should have required as false', async () => {
```

Comment:
This tests seems weird to me, we're adding the `required` to the the `element` and then doing the assertion to the same `element` is like is not testing anything.

Reply:
https://github.com/dotCMS/core-web/blob/96894a92b2299c428f66d30fdf342146475e256a/projects/dotcms-field-elements/src/components/dot-key-value/dot-key-value.e2e.ts#L58

Reply:
Then [this expect](https://github.com/dotCMS/core-web/pull/1008/files#diff-6952fd7925411bf96bde43ad187c18f6R57) is not neccesary right? and the description of the test should be like: "should add mark to the label" or something like that.